abci/abci.go

```diff
@@ -177,13 +206,26 @@ func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 			}
 
 			txSize := int64(len(txBz))
-			if totalTxBytes += txSize; totalTxBytes <= req.MaxTxBytes {
-				selectedTxs = append(selectedTxs, txBz)
-			} else {
+			if totalTxBytes += txSize; totalTxBytes > req.MaxTxBytes {
 				// We've reached capacity per req.MaxTxBytes so we cannot select any
 				// more transactions.
 				break selectTxLoop
 			}
+
+			// Gas Limit check
+			feeTx, ok := memTx.(sdk.FeeTx)
+			if !ok {
+				txsToRemove[memTx] = struct{}{}
+				continue selectTxLoop
+			}
+			gasLimit := feeTx.GetGas()
+			if totalGasLimit += int64(gasLimit); totalGasLimit > maxGasLimit {
```

Comment:
What happens if totalGasLimit is > maxGasLimit? In this case no more txs can be added? I.e we shld only be updating totalGasLimit after this check passes

Reply:
if we're approaching the max block gas limit, we break the second we find a tx that exceeds. this is probably fine for almost all cases, and the reason why i think breaking might be better is you'll otherwise have to iterate through the rest of the mempool to get a proposal (which eats up time). we can update afterwards since it seems you have a stronger preference for that, implemented it this way to maintain parity with the totalsize check.

Reply:
Kk